docs/ref/models/constraints.txt

```diff
@@ -69,3 +69,17 @@ date.
 .. attribute:: UniqueConstraint.name
 
 The name of the constraint.
+
+``condition``
+-------------
+
+.. attribute:: UniqueConstraint.condition
+
+A :class:`Q` object that specifies the condition you want the constraint to
+enforce.
+
+For example, ``UniqueConstraint(fields=['email'], condition=Q(status='ACTIVE')``
```

Comment:
I'm having trouble wrapping my head around this example. What model has these fields and how does this constraint make logical sense? Perhaps we can find a more obvious example.

Reply:
I think about next:

    class ResetEmail(models.Model):
        email = models.EmailField()
        status = model.CharField(choices=('ACTIVE', 'DONE', 'OUTDATED'))
        token =  model.CharField()
        class Meta:
            constraints = (
                # only one token can be used for email resetting at one time
                UniqueConstraint(fields=['email'], condition=Q(status='ACTIVE')),
            )

I also have another one:

    class DocumentBlock(models.Model):
        document = model.ForeignKey('Document')
        block_type = model.CharField(choices=('COVER', 'PARAGRAPH'))
        class Meta:
            constraints = (
                # document can have only one cover block
                UniqueConstraint(fields=['document'], condition=Q(block_type='COVER')),
            )

it's can be quite specific cases, so maybe someone can propose more clear example?

Reply:
How about:
```
For example, ``UniqueConstraint(fields=['user'], condition=Q(status='DRAFT')``
ensures that each user only has one draft.
```

Reply:
fine for me